### PR TITLE
Update X_NUCLEO_GNSS1A1_HelloWorld_UART.ino

### PR DESCRIPTION
**Pull request by altaga**
Library compatibility with LoRaWAN Development Board fixed - B-L072Z-LRWAN1

https://developer.helium.com/devices/devkit/lorawan-development-board

### CODE REVIEW

**Comment by cparata** [comment]
Hi @altaga ,
thanks for the PR, but your commits are not clear to me. It seems that you added some Arduino stuff on LRWAN and STM32 boards that should not be the case. I checked the schematics of the LRWAN board and the serial instance should be correct also for this platform. So, I suppose that the current code in the repository should work out-of-the-box also on LRWAN board, isn't it?
Best Regards,
Carlo

**Comment by altaga** [comment]
Right now im in a contest.

https://www.hackster.io/contests/IoTforGood

They sent us this hardware as a gift

https://developer.helium.com/devices/devkit

I tried to run the code on the board and it didn't work, but adding that if statement with that library already works perfectly.

#elif ARDUINO_STM32L0_B_L072Z_LRWAN1
#include <avr/dtostrf.h>

**Comment by fpistm** [comment]
> ARDUINO_STM32L0_B_L072Z_LRWAN1

Where does this definition come from?

**Comment by altaga** [comment]
The organizers of the contest have this documentation to program the board and to be able to program it with Arduino.

https://developer.helium.com/devices/arduino-quickstart/st-discovery-lrwan1

The definition was obtained directly from the documentation as seen in the image.

image

This is the official github for the board.

https://github.com/GrumpyOldPizza/ArduinoCore-stm32l0

**Comment by cparata** [comment]
Hi @fpistm ,
I guess he is using the porting of GrumpyOldPizza.github.io as you can see here:

https://developer.helium.com/devices/arduino-quickstart/st-discovery-lrwan1#software

Anyway, I guess that the current library is compatible with the porting of the Arduino Core.
But, I do not have issue to add this define to get the library compatible also with te GrumpyOldPizza core.
What do you think Frederic about that?
Best Regards,
Carlo

**Comment by altaga** [comment]
It is a small change to make the library more compatible with the board, since several people in the contest could benefit from this change, but if you do not consider that it is fine, there is no problem.

Without the IF

image

With the IF

image

**Comment by fpistm** [comment]
There is no issue to add the compatibility.
The main issue is that the define used is only for a board not for the core.
Looking at the platform.txt:
https://github.com/GrumpyOldPizza/ArduinoCore-stm32l0/blob/c845a3b4b90b8a8fe95c6fedd4fa927dae14ba45/platform.txt#L78

which define: `-DARDUINO_ARCH_{build.arch}`
with `build.arch=stm32l0`
So this should be:
`ARDUINO_ARCH_stm32l0`

@altaga could you test with this define, please?

**Comment by cparata** [comment]
Hi @altaga ,
I got your point. What I mean it is that there exist another working Arduino porting for the LRWAN board that is the one available in the "STM32duino Core" that you can install from this github repository as well. I'm pretty sure that with the "STM32duino Core" the library works without any issue. I did not know the GrumpyOldPizza core for the LRWAN board. Normally all the X-NUCLEO libraries are tested using the "STM32duino Core". Anyway, now it is clear to me the context and I do not have any issue to get this library also compatible with GrumpyOldPizza core. Anyway, we cannot guarantee support for this core in future if some bugs should arise because our reference is the "STM32duino Core".
Best Regards,
Carlo

**Comment by fpistm** [comment]
Moreover, this should be updated as well:
https://github.com/stm32duino/X-NUCLEO-GNSS1A1/blob/677450619ad52416998d6f1ee2165be055da97d7/library.properties#L9

to add `stm32l0` arch

**Comment by altaga** [comment]
The architecture appears correctly to me in the definition of the board, and when changing it in the code it still gives me an error.

image

image

Already added, restart the arduino and it still doesn't compile

image

**Comment by fpistm** [comment]
Well look in the log what is the correct value, it is possible it is uppercase:
`ARDUINO_ARCH_STM32L0`

**Comment by altaga** [comment]
image

Nope, not yet :/

**Comment by fpistm** [comment]
Please check in the log, if you don't have the log then enable the verbose in the IDE preference.
If the switch is correct and then it does not work then there is an issue with the core platform.txt which does not define it for all recipes.

**Comment by altaga** [comment]
Architecture is correct in the compilation.

image

**Comment by fpistm** [comment]
So this should work, but I think your issue come from the preproc syntax, this should be 

`#elif defined(ARDUINO_ARCH_STM32L0) `

because you want test the definition not the value.

**Comment by altaga** [comment]
With this definition work!
image

**Comment by cparata** [comment]
Ok, good.
So I will rework a bit the PR with the modifications suggested by @fpistm and then I will release a new version of the library. Thanks to all.
Best Regards,
Carlo

File: examples/X_NUCLEO_GNSS1A1_HelloWorld_UART/X_NUCLEO_GNSS1A1_HelloWorld_UART.ino [hunk 1]
```diff
@@ -66,6 +66,8 @@ HardwareSerial Serial1(PA10, PA9);
 
 #ifdef ARDUINO_SAM_DUE
 #include <avr/dtostrf.h>
+#elif ARDUINO_STM32L0_B_L072Z_LRWAN1
```
Comment (fpistm) [inline examples/X_NUCLEO_GNSS1A1_HelloWorld_UART/X_NUCLEO_GNSS1A1_HelloWorld_UART.ino:69]:
```suggestion
#elif defined(ARDUINO_ARCH_STM32L0)
```

**Review by fpistm — CHANGES_REQUESTED** [review]

**Review by fpistm — APPROVED** [review]

**Comment by fpistm** [comment]
@cparata 
I've made the change.
Simply do a "squash and merge" , then update the library.properties for arch and version.

**Comment by cparata** [comment]
Hi Frederic,
actually I would like also to update the I2C example that should work also with the other Core, so I prefer to rework a bit the PR on my tree and then I will upload the changes. Do I need to add also the "stm32l0" arch in the library file?
Best Regards,
Carlo

**Comment by fpistm** [comment]
@cparata 
OK no worry ;)
yes you could add "stm32l0" else when user install the library and want use it with @GrumpyOldPizza core this will be displayed as not compatible.

**Comment by cparata** [comment]
Ok, I merged the modifications in the master repo. I will publish a new release immediately. 
@altaga , you can check also the I2C examples; they should work also with the LRWAN board.
Best Regards,
Carlo

**Comment by fpistm** [comment]
Fix by #15

**Comment by altaga** [comment]
> Ok, I merged the modifications in the master repo. I will publish a new release immediately.
> @altaga , you can check also the I2C examples; they should work also with the LRWAN board.
> Best Regards,
> Carlo

Yes! i will check this examples!

**Comment by GrumpyOldPizza** [comment]
Guess, alternatively, I could quickly update the code on ArduinoCore-stm32l0 to support X_NUCLEO_GNSS1A1

The ARDUINO_STM32L0_B_L072Z_LRWAN1 give me moment to pause. That should really be ARDUINO_B_L072Z_LRWAN1. Guess I need to fix that on the weekend.

**Comment by cparata** [comment]
Hi @GrumpyOldPizza , I guess it is not needed anymore. I have already released a new version of the X-NUCLEO-GNSS1A1 library that should be compatible with your Core for LRWAN board. We refer to your Core using the "ARDUINO_ARCH_STM32L0" define, so if you want to update the define for B_L072Z_LRWAN1, the code should still work.
Best Regards,
Carlo

**Pull request closed without merging**